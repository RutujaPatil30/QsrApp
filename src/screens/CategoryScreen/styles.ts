import {StyleSheet} from 'react-native';
import theme from '../../utils/themes';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';



export const styles = StyleSheet.create({
  container: {
    flex:1,
    backgroundColor: theme.colors.backgroundGrey,
    paddingVertical:hp('2%'),
    
  },
  flatlistContainer: {
    width: wp('95%'),
    alignSelf: 'center',
    padding: hp('1%'),
  },
});
