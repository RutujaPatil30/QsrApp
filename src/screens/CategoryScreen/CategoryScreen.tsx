import React from 'react';
import {FlatList, View} from 'react-native';
import {useDispatch, useSelector} from 'react-redux';
import CustomCategoryElements from '../../components/CustomCategoryElements/CustomCategoryElements';
import {add_to_cart_request} from '../../redux/action'
import {styles} from './styles';
import ProductDetails from '../../utils/productDetails'

const CategoryScreen = (props: any) => {
  const {navigation, route} = props;

  const dispatch = useDispatch();
  const userId = useSelector((state: any) => state.reducer.currentUser.id);
  const cartItems = useSelector((state: any) => state.reducer.cartItems);
  const AddItems = (item: any) => {
    dispatch(add_to_cart_request(userId, item));
  };

  return (
    <View style={styles.container}>
      <FlatList
        //@ts-ignore
        data={ProductDetails[route?.params?.title]}
        renderItem={({item}) => (
          <View style={styles.flatlistContainer}>
            <CustomCategoryElements
              testID='CategoryScreen_Elements'
              itemImage={item.image}
              itemName={item.title}
              itemDescription={item.description}
              itemPrice={item.price}
              onPressAdd={() => {
                AddItems(item);
              }}
              onPress={() => {
                navigation.navigate('ProductDetailsScreen', {
                  image: item.image,
                  name: item.title,
                  description: item.description,
                  price: item.price,
                  itemDetails: item,
                });
              }}
            />
          </View>
        )}
        showsVerticalScrollIndicator={false}
      />
    </View>
  );
};

export default CategoryScreen;
