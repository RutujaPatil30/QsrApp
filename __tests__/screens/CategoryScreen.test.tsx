
import React from 'react';
import { render, screen } from '@testing-library/react-native';
import CategoryScreen from '../../src/Screens/CategoryScreen/CategoryScreen';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

const mockStore = configureStore([]);
const store = mockStore({
    reducer: {
        cartItems: [], 
    },
});

const mockRoute = {
    params: {
      title: 'Breakfast', 
    },
  }

describe('Checking Category Screen ', () => {
    it('should render the expected UI elements', async () => {
        render(
          <Provider store={store}>
            <CategoryScreen route={mockRoute}  />
          </Provider>
        );
      
        const categoryElements = await screen.findAllByTestId('CategoryScreen_Elements');
      
        expect(categoryElements.length).toBeGreaterThan(0);
      });
});

describe('CategoryScreen Snapshot Test', () => {
  it('renders correctly', () => {
      const tree = renderer.create(
          <Provider store={store}>
              <CategoryScreen />
          </Provider>
      ).toJSON();
      expect(tree).toMatchSnapshot();
  });
});