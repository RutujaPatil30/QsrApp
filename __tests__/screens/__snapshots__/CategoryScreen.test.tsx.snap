// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryScreen Snapshot Test renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#f6f6f6",
      "flex": 1,
      "paddingVertical": 26.5,
    }
  }
>
  <RCTScrollView
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
</View>
`;
